js/layout-javascript/small-card-code.js

```diff
@@ -179,13 +181,18 @@ DynamicList.prototype.attachObservers = function() {
         'desc': 'asc'
       };
 
+      _this.sortField = $sortListItem.data('sortField');
+      _this.sortOrder = newSortOrder[oldSortOrder];
       _this.resetSortIcons();
+      _this.data.forceRenderList = true;
 
-      $sortOrderIcon.removeClass(removeClasses[sortOrder]).addClass(addClasses[sortOrder]);
-      $sortListItem.data('sortOrder', newSortOrder[sortOrder]);
+      $sortOrderIcon.removeClass(removeClasses[oldSortOrder]).addClass(addClasses[oldSortOrder]);
+      $sortListItem.data('sortOrder', newSortOrder[oldSortOrder]);
 
-      _this.modifiedListItems = _.orderBy(_this.modifiedListItems, sortField, newSortOrder[sortOrder]);
-      _this.renderLoopHTML();
+      _this.searchData()
```

Comment:
If you pass the parameters through to the function, you can achieve the following:

1. Use the presence of `sortField` (assuming `sortOrder` defaults to `ASC`) in `searchData()` to determine the list needs to be re-rendered.
1. You won't need to set up `this.sortField` and `this.sortOrder`

```suggestion
      _this.searchData({
        sortField,
        sortOrder
      })
```